test/src/test/java/org/corfudb/runtime/collections/CorfuQueueTest.java

```diff
@@ -67,6 +68,20 @@ public void basicQueueOrder() {
         assertThat(records.get(0).getRecordId().compareTo(records2.get(1).getRecordId())).isLessThan(0);
     }
 
+    @Test
+    public void queueWithSecondaryIndexCheck() {
```

Comment:
can you write more tests and cover more border cases if they exist?

Reply:
right now it's only the serializer which is settable in this PR, so I only have unit test for that. Later if we add more features with queries, then I will add tests along with that feature.